### ISSUE TITLE
Change masks in ldtr checks

### ISSUE THREAD

**Issue by connojd** [Bareflank/hypervisor#224]
We should use `SEGMENT_ACCESS_RIGHTS_RESERVED` rather than the hardcoded `0x0F00` in `vmcs_intel_x64::check_guest_ldtr_access_rights_reserved_must_be_0()`.

In `vmcs_intel_x64::check_guest_ldtr_access_rights_remaining_reserved_bit_0()`, the "remaining" reserved bits are 31:17 so we should have 

``` cpp
if ((ldtr_access & 0xFFFE0000) != 0)
    throw std::logic_error("guest ldtr access rights bits 31:17 must be 0 if ldtr is usable");
```

rather than

``` cpp
if ((ldtr_access & SEGMENT_ACCESS_RIGHTS_RESERVED) != 0)
    throw std::logic_error("ldtr access rights reserved bits must be 0 if ldtr is usable");
```

**Comment by rianquinn** [comment]
I am confused by the explanation here (sorry I've been really sick this week). What does the code do now, and what do you think it should look like?

**Comment by connojd** [comment]
I've been sick too :(. Ok there are two functions with an issue each.  We have this `#define`d from intrinsics_x64.h:  `SEGMENT_ACCESS_RIGHTS_RESERVED 0x0F00`.  In `check_guest_ldtr_access_rights_reserved_must_be_0`, we have
```cpp
...
 if ((ldtr_access & 0x0F00) != 0)
    throw ...
...
```
when it should be this 
```cpp
if ((ldtr_access & SEGMENT_ACCESS_RIGHTS_RESERVED) != 0)
    throw ...
```
so just a nitpicky typographical issue there. 

In `check_guest_ldtr_access_rights_remaining_reserved_bit_0`, right now we have
```cpp
...
if ((ldtr_access & SEGMENT_ACCESS_RIGHTS_RESERVED) != 0)
    throw ...
...
```
but according to section 26.3.1.2 (very last bullet), the bits that need to be `0` are 31:17 of ldtr_access, so the check would need to be
```cpp
...
if ((ldtr_access & 0xFFFE0000) != 0)
    throw ...
...
```

**Comment by rianquinn** [comment]
Ok... makes sense to me.

**Comment by connojd** [comment]
Fixed in #343
